app/src/main/java/com/frinder/frinder/activity/DiscoverActivity.java

```diff
@@ -101,74 +114,109 @@ private void getCurrentLocation() {
 
-                    SharedPreferences pref = PreferenceManager.getDefaultSharedPreferences(this);
-                    int radius = pref.getInt("radius", 200);
-                    if (distance <= radius) {
-                        Log.d(TAG, user.getName() + ", distance = " + distance);
-                        nearbyUsers.add(user);
-                        distanceFromCurrentUser.add(distance);
-                        distances.add(distance);
+        discoverFirebaseDas.getNearbyUsers(currentUser, searchRadius, new DiscoverFirebaseDas.OnCompletionListener() {
+            @Override
+            public void onNearbyUserListReceived(ArrayList<String> userIdList) {
+                if (userIdList != null) {
+                    unFilteredNearbyUsersCount = userIdList.size();
+                    for (String userId : userIdList) {
+                        discoverFirebaseDas.getUser(userId, new DiscoverFirebaseDas.OnCompletionListener() {
+                            @Override
+                            public void onUserReceived(User user) {
+                                Log.d(TAG, "Checking if user " + user.getName() + " matches filters");
+                                //check if user matches filters
+
+                                //TODO uncomment when discoverable field is included
+                                boolean isDiscoverable = user.getDiscoverable();
+
+                                boolean correctTimestamp = false;
+                                Date currentTimestamp = new Date();
+                                if (currentTimestamp.getTime() - user.getTimestamp().getTime() < (TIME_USER_LAST_ACTIVE * 60000)) {
+                                    correctTimestamp = true;
+                                }
+
+                                Log.d(TAG, user.getName() + ", user interests =" + user.getInterests().toString());
+                                Log.d(TAG, user.getName() + ", filter interests =" + filterInterests.toString());
+                                boolean interestMatch = false;
+                                if (filterInterests.size() > 0) {
+                                    if (user.getInterests() != null && user.getInterests().size() > 0) {
+                                        ArrayList<String> filterInterestsCopy = new ArrayList<String>(filterInterests);
+                                        filterInterestsCopy.retainAll(user.getInterests());
+                                        if (filterInterestsCopy.size() > 0) interestMatch = true;
+                                        Log.d(TAG, user.getName() + ", common interests =" + filterInterestsCopy.toString());
+                                    }
+                                }
+
+                                //TODO uncomment line below and comment the next one when discoverable field is included
+                                if (isDiscoverable && correctTimestamp && interestMatch) {
```

Comment:
Did you mean to comment this line? Ignore if this is what you intended.

Reply:
I need to remove that TODO comment there. It's already done and works.